lapack/testlapack/dlagv2.go

```diff
@@ -0,0 +1,182 @@
+	// r2 = [ CSR -SNR; SNR CSR]
+	r2 := blas64.General{
+		Data: []float64{csr, -snr, snr, csr},
+		Rows: 2, Cols: 2, Stride: 2,
+	}
+	isComplexEig := wi1 != 0
+	name := fmt.Sprintf("cmplx=%v,lda=%d,ldb=%d,aKind=%d,bKind=%d", isComplexEig, lda, ldb, aKind, bKind)
+	// aOrigStr := sprint2x2Block("A", aCopy.Data, aCopy.Stride, 4)
+	// bOrigStr := sprint2x2Block("B", bCopy.Data, bCopy.Stride, 4)
+
+	aStr := sprint2x2Block("As", a.Data, a.Stride, 4)
+	bStr := sprint2x2Block("Bs", b.Data, b.Stride, 4)
+
+	if wi1 < 0 {
+		t.Fatalf("%s: negative wi1; wi1=%g,wi2=%g,\n%s\n%s", name, wi1, wi2, aStr, bStr)
+		return
+	}
+	if isComplexEig && wi2 != -wi1 {
+		t.Fatalf("%s: wi1 != -wi2; wi1=%g,wi2=%g,\n%s\n%s", name, wi1, wi2, aStr, bStr)
+		return
+	}
+	if math.Abs(b.Data[b.Stride]) > tol {
+		t.Fatalf("%s: expected b to remain upper triangular:\n%s", name, bStr)
+		return
+	}
+	if isComplexEig && b.Data[0] < b.Data[b.Stride+1] {
+		t.Errorf("%s: expected b diagonal elements to be in descending order:\n%s", name, bStr)
+	}
+
+	if !isComplexEig && !isSchurCanonicalGeneral(a, tol) {
+		t.Fatalf("%s: a is not Schur canonical:\n%s", name, aStr)
+		return
+	}
+
+	if !isSchurCanonicalGeneral(b, tol) {
+		t.Fatalf("%s: b is not Schur canonical:\n%s", name, bStr)
+		return
+	}
+
+	switch {
+	case isComplexEig:
+		// Complex eigenvalue pair.
+		if wr1 != wr2 {
+			t.Fatalf("%s: complex eigenvalue but wr1 != wr2; wr1=%g, wr2=%g,\n%s\n%s", name, wr1, wr2, aStr, bStr)
+			return
+		}
+		if beta1 != beta2 {
+			t.Fatalf("%s: complex eigenvalue but scale1 != scale2; scale1=%g, scale2=%g,\n%s\n%s", name, beta1, beta2, aStr, bStr)
+			return
+		}
+		if b.Data[1] != 0 {
+			t.Errorf("%s: expected b to be diagonal on complex pair:\n%s", name, bStr)
+		}
+	default:
+		// Real eigenvalue pair.
+		if wi1 != 0 || wi2 != 0 {
+			t.Fatalf("%s: real eigenvalue but wi1 != 0 or wi2 != 0; wi1=%g, wi2=%g,\n%s\n%s", name, wi1, wi2, aStr, bStr)
+			return
+		}
+		if a.Data[a.Stride] != 0 {
+			t.Errorf("%s: expected a to be upper triangular on real pair:\n%s", name, aStr)
+		}
+	}
+
+	// Test Rotations.
+
+	aux := nanGeneral(2, 2, 2)
+	result := nanGeneral(2, 2, 2)
+	// Aschur = r1 * A * r2
+	// Bschur = r1 * B * r2
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, r1, aCopy, 0, aux)
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, aux, r2, 0, result)
+	if !equalApproxGeneral(result, a, tol) {
+		t.Errorf("%s: bad rotation:\n%s\n%s",
+			name, aStr, sprint2x2Block("A", result.Data, result.Stride, 4))
+	}
+
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, r1, bCopy, 0, aux)
+	blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, aux, r2, 0, result)
+	if !equalApproxGeneral(result, b, tol) {
+		t.Errorf("%s: bad rotation:\n%s\n%s",
+			name, bStr, sprint2x2Block("B", result.Data, result.Stride, 4))
+	}
+
+	// Test eigenvalues.
+
+	res, err := residualDlag2(aCopy, bCopy, beta1, complex(wr1, wi1))
+	if err != nil {
+		// t.Logf("%s: invalid input data: %v\n%s\n%s\n%s\n%s", name, err, aStr, bStr, aOrigStr, bOrigStr)
```

Comment:
Please clean up. And why silently return? Is the error an indication of a bad input data for the test?